packages/itwinui-react/src/core/Table/Table.tsx

```diff
@@ -656,6 +656,12 @@ export const Table = <
     }
   }
 
+  if (data.some((item) => (item.subRows as T[]).length > 0) && subComponent) {
```

Comment:
Can you also add a check for `process.env.NODE_ENV === 'development'`?

Reply:
I have added the if condition to check for the dev environment.

Reply:
it should be made part of the same `if` condition instead of a nested one.

Reply:
I have made the change.